source/overloading.tex

```diff
@@ -3540,24 +3528,25 @@
 \indextext{overloading!unary operator}
 
 \pnum
-A prefix unary operator shall be implemented by a
-non-static member function\iref{class.mfct} with no parameters or a
-non-member function with one parameter.
+A \defnadj{prefix unary}{operator function}
+is a function named \tcode{\keyword{operator}@}
+for a prefix unary operator \tcode{@}\iref{expr.unary.op} that is either
+a non-static member function\iref{class.mfct} with no parameters or
+a non-member function with one parameter.
 \indextext{unary operator!interpretation of}%
-Thus, for any prefix unary operator
-\tcode{@},
-\tcode{@x}
-can be interpreted as either
-\tcode{x.op\-er\-a\-tor@()}
-or
-\tcode{operator@(x)}.
-If both forms of the operator function have been declared,
-the rules in~\ref{over.match.oper} determine which, if any, interpretation is
-used.
-See~\ref{over.inc} for an explanation of the postfix unary operators
-\tcode{++}
-and
-\tcode{\dcr}.
+For a \grammarterm{unary-expression}
+of the form \tcode{@ \grammarterm{cast-expression}},
+the operator function is selected by overload resolution\iref{over.match.oper}.
+If a member function is selected,
+the expression is interpreted as
+\tcode{\grammarterm{cast-expression} . \keyword{operator}@()};
+otherwise, if a non-member function is selected,
+the expression is interpreted as
+\tcode{\keyword{operator}@( \grammarterm{cast-expression} )}.
+\begin{note}
+The operators \tcode{++} and \tcode{\dcr}\iref{expr.pre.incr}
+are described in~\ref{over.inc}.
+\end{note}
```

Comment:
Hmm. They're actually described in this section too, and the rules are (fortunately) identical.
We should make these orthogonal: either `operator--` and `operator++` are not prefix unary operator functions, or the section on increment / decrement should instead describe postfix unary operator functions and not repeat the description of prefix `--` and `++`. What do you think?

Reply:
I think we should only describe postfix ++ and -- as special. I should also apply the grammar rewriting to that section so that we can show an actual call to `operator++(0)`.

Reply:
Ah, no, that's wrong. Increment/decrement is not a _unary-operator_ per the grammar; it's something special. I am now highlighting that by using the \grammarterm.

Pushing for further review.

Reply:
This approach seems fine to me.